mono/metadata/assembly.c

```diff
@@ -428,7 +428,7 @@ encode_public_tok (const guchar *token, gint32 len)
 gboolean
 mono_public_tokens_are_equal (const unsigned char *pubt1, const unsigned char *pubt2)
 {
-	return memcmp (pubt1, pubt2, 16) == 0;
+	return g_ascii_strncasecmp ((const char*) pubt1, (const char*) pubt2, 16) == 0;
```

Comment:
The previous case code handled embedded nuls, and always compared all 16 bytes.
Please consider add `g_memicmp`. On Windows it will just call `_memicmp`.
And then as a bonus, you won't need the casts.

Reply:
Well, hey, that's weird/good, g_ascii_strncasecmp doesn't stop at a terminal nul. So I guess nevermind.

Reply:
Oh, `g_ascii_strncasecmp` does not stop at nuls I guess so somewhat nevermind.